### PR TITLE
Workshop two

### PR DESCRIPTION
**Pull request by chrisolsen**
Let me know if anything needs to be added/removed or changed.

### CODE REVIEW

**Comment by chrisolsen** [comment]
I think things are ready right now. I have gone through it copying all the code snippets and things work, so if you are busy that is probably ok. Have you been able to look at the PR for the darksky library. The code that I have right now for the presentation is dependent on the PR request that I sent. If you can merge it that would be great, if not let me know and I will change things on my side to make things work.

**Comment by chrisolsen** [comment]
I think things are ready right now. I have gone through it copying all the code snippets and things work, so if you are busy that is probably ok. Have you been able to look at the PR for the darksky library. The code that I have right now for the presentation is dependent on the PR request that I sent. If you can merge it that would be great, if not let me know and I will change things on my side to make things work.

**Comment by chrisolsen** [comment]
I think things are ready right now. I have gone through it copying all the code snippets and things work, so if you are busy that is probably ok. Have you been able to look at the PR for the darksky library. The code that I have right now for the presentation is dependent on the PR request that I sent. If you can merge it that would be great, if not let me know and I will change things on my side to make things work.

**Comment by nathany** [comment]
Merged your Darksky PR. 

Sorry for the delay, I have GitHub set to only email me when mentioned, otherwise I'd get too many emails from other projects I'm watching.

**Comment by chrisolsen** [comment]
I am converting the markdown to a go presentation, so I might toss this branch and submit another in a bit. So don't merge this one :)

File: 2018-10/workshop-two/handler.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	case "text/html":
+		if len(h.viewPaths) == 0 {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "at least one view must be passed in")
+			return
+		}
+		tmpl, err := h.templates.get(h.viewPaths, h.helpers)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, fmt.Sprintf("template new failure: %v", err))
```
Comment (nathany) [inline 2018-10/workshop-two/handler.go]: No need to use Sprintf inside Fprintf. It does that for you. Same below.

File: 2018-10/workshop-two/main.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"os"
+
+	"github.com/edmontongo/darksky"
+)
+
+func main() {
+	var secret string
+	// define and parse out command line flags
+	flag.StringVar(&secret, "secret", "", "DarkSky API key (required)")
+	port := flag.String("port", "3001", "Port the service will be listening on")
```
Comment (nathany) [inline 2018-10/workshop-two/main.go]: are you intentionally using two different methods of getting flags (String vs. StringVar)?

File: 2018-10/workshop-two/workshop-two.md [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+
+1. https://github.com/codegangsta/gin
+
+```bash
+go get github.com/codegangsta/gin
+```
+
+---
+
+# Hello World
```
Comment (nathany) [inline 2018-10/workshop-two/workshop-two.md]: do we need this?

File: 2018-10/workshop-two/workshop-two.md [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+lat, err := strconv.ParseFloat(q.Get("lat"), 10)
+if err != nil {
+    http.Error(w, "Invalid lat querystring param", http.StatusBadRequest)
+    return
+}
+```
+
+## Exercise
+
+Parse out the longitude value as well, then test it via the browser
```
Comment (nathany) [inline 2018-10/workshop-two/workshop-two.md]: Maybe add a gist for the code on the slide, or do you want people to type it all out?

File: 2018-10/workshop-two/workshop-two.md [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+{{end}}
+```
+
+
+---
+
+# Refactor
+
+1. Remove global darkskySvc
+2. Cache templates (won't cover this, but checkout the code on github)
```
Comment (nathany) [inline 2018-10/workshop-two/workshop-two.md]: Maybe we could add this server code to the darksky repo for people to play with -- perhaps not until after the meetup though, given that it's already Monday.

**Comment by nathany** [comment]
Thanks Chris.

File: 2018-10/workshop-two/handler.go (hunk 1 above)
Comment (chrisolsen) [reply 2018-10/workshop-two/handler.go]: Good catch

File: 2018-10/workshop-two/workshop-two.md (hunk 4 above)
Comment (chrisolsen) [reply 2018-10/workshop-two/workshop-two.md]: I really should've had gists for this presentation; next time :)

**Pull request closed without merging**